serverless.yml

```diff
@@ -14,37 +16,43 @@ provider:
           Resource: "*"
         - Effect: Allow
           Action:
-            - 's3:ListBucket'
-            - 's3:GetObject'
-            - 's3:GetObjectAcl'
-            - 's3:PutObject'
-            - 's3:PutObjectAcl'
-            - 's3:ReplicateObject'
-            - 's3:DeleteObject'
+            - "s3:ListBucket"
+            - "s3:GetObject"
+            - "s3:GetObjectAcl"
+            - "s3:PutObject"
+            - "s3:PutObjectAcl"
+            - "s3:ReplicateObject"
+            - "s3:DeleteObject"
           Resource:
-            - arn:aws:s3:::${env:SLACK_INSTALLATION_S3_BUCKET_NAME}
-            - arn:aws:s3:::${env:SLACK_INSTALLATION_S3_BUCKET_NAME}/*
-            - arn:aws:s3:::${env:SLACK_STATE_S3_BUCKET_NAME}
-            - arn:aws:s3:::${env:SLACK_STATE_S3_BUCKET_NAME}/*
-            - arn:aws:s3:::${env:OPENAI_S3_BUCKET_NAME}
-            - arn:aws:s3:::${env:OPENAI_S3_BUCKET_NAME}/*
+            - arn:aws:s3:::${self:custom.SLACK_STATE_S3_BUCKET_NAME}
+            - arn:aws:s3:::${self:custom.SLACK_STATE_S3_BUCKET_NAME}/*
+            - arn:aws:s3:::${self:custom.SLACK_INSTALLATION_S3_BUCKET_NAME}
+            - arn:aws:s3:::${self:custom.SLACK_INSTALLATION_S3_BUCKET_NAME}/*
+            - arn:aws:s3:::${self:custom.OPENAI_S3_BUCKET_NAME}
+            - arn:aws:s3:::${self:custom.OPENAI_S3_BUCKET_NAME}/*
   environment:
-    SERVERLESS_STAGE: ${opt:stage, 'prod'}
     SLACK_SIGNING_SECRET: ${env:SLACK_SIGNING_SECRET}
     SLACK_CLIENT_ID: ${env:SLACK_CLIENT_ID}
     SLACK_CLIENT_SECRET: ${env:SLACK_CLIENT_SECRET}
     SLACK_SCOPES: ${env:SLACK_SCOPES}
-    SLACK_INSTALLATION_S3_BUCKET_NAME: ${env:SLACK_INSTALLATION_S3_BUCKET_NAME}
-    SLACK_STATE_S3_BUCKET_NAME: ${env:SLACK_STATE_S3_BUCKET_NAME}
-    OPENAI_S3_BUCKET_NAME: ${env:OPENAI_S3_BUCKET_NAME}
-    OPENAI_TIMEOUT_SECONDS: 25
+    SLACK_INSTALLATION_S3_BUCKET_NAME: ${self:custom.SLACK_INSTALLATION_S3_BUCKET_NAME}
+    SLACK_STATE_S3_BUCKET_NAME: ${self:custom.SLACK_STATE_S3_BUCKET_NAME}
+    ADMIN_SLACK_IDS: ${env:ADMIN_SLACK_IDS}
+    OPENAI_MODEL: "gpt-3.5-turbo"
+    OPENAI_TEMPERATURE: 0.5
+    OPENAI_MAX_TOKENS: 3072
+    OPENAI_S3_BUCKET_NAME: ${self:custom.OPENAI_S3_BUCKET_NAME}
+    OPENAI_TIMEOUT_SECONDS: 30
     SLACK_APP_LOG_LEVEL: WARN
     TRANSLATE_MARKDOWN: true
+    # Get Bing auth_cookies by login bing in browser and using javascript "cookieStore.get("_U").then(result => console.log(result.value))"
+    BING_AUTH_COOKIE: ${env:BING_AUTH_COOKIE}
```

Comment:
minus the bing changes, I think there are good improvements overall to serverless.yaml. Not an expert but hope these get incorporated.